carla_ros_bridge/src/carla_ros_bridge/actor_factory.py

```diff
@@ -119,10 +119,15 @@ def update_available_objects(self):
 
         # update objects for pseudo actors here as they might have an carla actor as parent ######
         with self.spawn_lock:
+            task_queue = queue.Queue()
             while not self._task_queue.empty():
                 task = self._task_queue.get()
-                if task[0] == ActorFactory.TaskType.SPAWN_ACTOR and not self.node.shutdown.is_set():
-                    carla_actor = self.world.get_actor(task[1][0])
+                actor_id = task[1][0]
```

Comment:
There is an issue with actor_id. For `task[0] == ActorFactory.TaskType.DESTROY_ACTOR` case, the actor_id should be task[1], which task[1][0] does not exist. Add a if-else or try-except will fix it.